src/lib.rs

```diff
@@ -582,86 +703,302 @@ where
+where
+    C: Blockchain,
+    D: BatchDatabase,
+{
+    match online_subcommand {
+        Sync { max_addresses } => {
+            maybe_await!(wallet.sync(log_progress(), max_addresses))?;
+            Ok(json!({}))
+        }
+        Broadcast { psbt, tx } => {
+            let tx = match (psbt, tx) {
+                (Some(psbt), None) => {
+                    let psbt = base64::decode(&psbt).unwrap();
+                    let psbt: PartiallySignedTransaction = deserialize(&psbt).unwrap();
+                    psbt.extract_tx()
+                }
+                (None, Some(tx)) => deserialize(&Vec::<u8>::from_hex(&tx).unwrap()).unwrap(),
+                (Some(_), Some(_)) => panic!("Both `psbt` and `tx` options not allowed"),
+                (None, None) => panic!("Missing `psbt` and `tx` option"),
+            };
+
+            let txid = maybe_await!(wallet.broadcast(tx))?;
+            Ok(json!({ "txid": txid }))
+        }
+    }
+}
+
+#[cfg_attr(not(doc), allow(missing_docs))]
+#[cfg_attr(
+    doc,
+    doc = r#"
+Key sub-command
+
+Provides basic key operations that are not related to a specific wallet such as generating a
+new random master extended key or restoring a master extended key from mnemonic words.
+
+These sub-commands are **EXPERIMENTAL** and should only be used for testing. Do not use this
+feature to create keys that secure actual funds on the Bitcoin mainnet.  
+"#
+)]
+#[derive(Debug, StructOpt, Clone, PartialEq)]
+#[structopt(rename_all = "snake")]
+pub enum KeySubCommand {
+    /// Generates new random seed mnemonic phrase and corresponding master extended key
+    Generate {
+        /// Entropy level based on number of random seed mnemonic words
+        #[structopt(
+        name = "WORD_COUNT",
+        short = "e",
+        long = "entropy",
+        default_value = "24",
+        possible_values = &["12","24"],
+        )]
+        word_count: usize,
+        /// Seed password
+        #[structopt(name = "PASSWORD", short = "p", long = "password")]
+        password: Option<String>,
+    },
+    /// Restore a master extended key from seed backup mnemonic words
+    Restore {
+        /// Seed mnemonic words, must be quoted (eg. "word1 word2 ...")
+        #[structopt(name = "MNEMONIC", short = "m", long = "mnemonic")]
+        mnemonic: String,
+        /// Seed password
+        #[structopt(name = "PASSWORD", short = "p", long = "password")]
+        password: Option<String>,
+    },
+}
+
+/// Execute a key sub-command
+///
+/// Key sub-commands are described in [`KeySubCommand`].
+pub fn handle_key_subcommand(
+    network: Network,
+    subcommand: KeySubCommand,
+) -> Result<serde_json::Value, Error> {
+    let secp = Secp256k1::new();
+
+    match subcommand {
+        KeySubCommand::Generate {
+            word_count,
+            password,
+        } => {
+            let mnemonic_type = match word_count {
+                12 => MnemonicType::Words12,
+                _ => MnemonicType::Words24,
```

Comment:
Maybe return an error if an invalid number is passed? I think it's fine to only support 12 and 24, but if I pass like 7 I'd like to see an error rather than it silently use 24.

Reply:
Actually this is taken care of by `structopt`, if I pass an invalid value like 7 I get this message:
```
cargo run --all-features key generate -e 7
    ...
error: '7' isn't a valid value for '--entropy <WORD_COUNT>'
        [possible values: 12, 24]

```